data/src/test/java/com/linkedin/data/schema/resolver/TestDataSchemaResolver.java

```diff
@@ -460,22 +697,36 @@ public void testMapDataSchemaResolver()
                         "typeref has a circular reference to itself"
                     }
                 }
+        },
+        {
+            "Circular reference involving typerefs, using alias",
+            "pdl",
+            asMap(buildSystemIndependentPath("a1", "typeref1.pdl"), "@aliases = [\"typerefAlias1\"] typeref typeref1 = typeref2",
+                  buildSystemIndependentPath("a1", "typeref2.pdl"), "typeref typeref2 = typerefAlias1"
+                 ),
+            new String[][]
+                {
+                    {
+                        "typeref1",
+                        ERROR,
+                        "typeref has a circular reference to itself"
+                    }
+                }
         }
     };
   }
 
   @Test(dataProvider = "circularReferenceData")
-  public void testCircularReferences(String desc, Map<String, String> testSchemas, String[][] testLookupAndExpectedResults)
+  public void testCircularReferences(String desc, String extension, Map<String, String> testSchemas, String[][] testLookupAndExpectedResults)
   {
     boolean debug = false;
 
     for (String[] testLookupAndExpectedResult : testLookupAndExpectedResults)
     {
-      DataSchemaResolver resolver = new MapDataSchemaResolver(
-          SchemaParserFactory.instance(),
-          Arrays.asList(buildSystemIndependentPath("a1")),
-          ".pdsc", testSchemas);
-      lookup(resolver, new String[][] { testLookupAndExpectedResult}, File.separatorChar, debug);
+      DataSchemaResolver schemaResolver = new MapDataSchemaResolver(
+          extension.equals("pdsc") ? SchemaParserFactory.instance() : PdlSchemaParserFactory.instance(),
```

Comment:
Can we use MultiFormatDataSchemaResolver instead?

Reply:
MapDataSchemaResolver is written for this test cases so that is why I want to re-use them